odoo/tests/test-uninstalls.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+import argparse
+import logging
+import os
+import sys
+
+sys.path.append(os.path.abspath(os.path.join(__file__,'../../../')))
+
+import odoo
+from odoo.tools import topological_sort
+from odoo.netsvc import init_logger
+
+_logger = logging.getLogger(__name__)
+
+BLACKLIST = {
+    'auth_ldap', 'document_ftp', 'base_gengo', 'website_gengo', 'website_instantclick', 'pad',
+    'pad_project', 'note_pad', 'pos_cache', 'pos_blackbox_be', 'payment_test',
+}
+IGNORE = ('hw_', 'theme_', 'l10n_', 'test_', 'payment_')
+
+
+def install(db_name, module_id, module_name):
+    with odoo.api.Environment.manage():
+        with odoo.registry(db_name).cursor() as cr:
+            env = odoo.api.Environment(cr, odoo.SUPERUSER_ID, {})
+            module = env['ir.module.module'].browse(module_id)
+            module.button_immediate_install()
+    _logger.info('%s installed', module_name)
+
+
+def uninstall(db_name, module_id, module_name):
+    with odoo.api.Environment.manage():
+        with odoo.registry(db_name).cursor() as cr:
+            env = odoo.api.Environment(cr, odoo.SUPERUSER_ID, {})
+            module = env['ir.module.module'].browse(module_id)
+            module.button_immediate_uninstall()
+    _logger.info('%s uninstalled', module_name)
+
+
+def cycle(db_name, module_id, module_name):
+    install(db_name, module_id, module_name)
+    uninstall(db_name, module_id, module_name)
+    install(db_name, module_id, module_name)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Script for testing the install / uninstall / reinstall cycle of Odoo modules")
+    parser.add_argument("--database", "-d", type=str, required=True,
+        help="The database to test (note: must have only 'base' installed)")
+    parser.add_argument("--skip", "-s", type=str, nargs=1,
+        help="Comma-separated list of modules to skip (they will only be installed)")
+    parser.add_argument("--resume", "-r", type=str, nargs=1,
+        help="Skip the cycle on all modules up to the specified one")
```

Comment:
```suggestion
        help="Skip modules (only install) up to the specified one in topological order")
```
or something else that indicates the order